### PR TITLE
[KEP-3041] - remove node special feature and node alpha feature from test-infra for non release branch

### PR DESCRIPTION
**Pull request by kannon92**
Address the test-infra part of https://github.com/kubernetes/test-infra/issues/33823.

### CODE REVIEW

**Comment by kannon92** [comment]
/hold

We should merge this once 1.33 opens up for development.

**Review by AnishShah — COMMENTED** [review]
Thanks for doing this as it is confusing me to me too.

File: config/jobs/kubernetes/sig-node/containerd.yaml
Comment (AnishShah) [inline config/jobs/kubernetes/sig-node/containerd.yaml:230]: Instead of removing, should we replace it with specific feature labels like `Feature:PodLevelResource` and `Feature:InPlacePodVerticalScaling`? This would allow us to exclude tests related to in-development features from certain jobs.

Comment (AnishShah) [reply config/jobs/kubernetes/sig-node/containerd.yaml:230]: During the sig-node CI meeting, we discussed a strategy to submit your 2 changes. This approach would streamline the PR submission process. By replacing the generic labels NodeAlphaFeature and NodeSpecialFeature with specific feature labels, we can prioritize the submission of this PR, followed by the k/k PR.

Comment (kannon92) [reply config/jobs/kubernetes/sig-node/containerd.yaml:230]: That was about a different item. In this case, We are skipping based on NodeSpecialFeature. I think we can drop this as is.

Comment (kannon92) [reply config/jobs/kubernetes/sig-node/containerd.yaml:230]: In this case I think merging the k8s PR first and then this one is sufficient..

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
